grails-app/services/com/unifina/service/CanvasService.groovy

```diff
@@ -211,6 +211,8 @@ class CanvasService {
 				case ExampleType.COPY:
 					Canvas c = new Canvas()
 					InvokerHelper.setProperties(c, example.properties)
+					c.dashboardItems = new HashSet<DashboardItem>()
+					c.permissions = new HashSet<Permission>()
```

Comment:
InvokeHelper creates some funny references and hibernate dies. Here is a fix, however I cannot remember if Hibernate has a problem with this? It is tested and works with example canvas copy.

Reply:
This is a bit of a time bomb... If we add new one-to-many associations to `Canvas`, this will break again (and prevent new users from signing up!). 

The problem might be that `InvokerHelper` is dumb enough to try and copy the associations of `example` to the new `Canvas`, which of course can't work at all.

Have you tried `Canvas c = new Canvas(example.properties)`? This shouldn't try to bind unbindable properties (like `id`), but I'm not sure how it handles those associative `Set`s.